pool/pool.go

```diff
@@ -64,11 +71,13 @@ func getIndex(a []*PooledTransaction, target time.Time, low, high int) int {
 // Set inserts a transaction into the pool, returning
 // true if the Transaction was inserted (was valid).
 func (p *Pool) Set(t *blockchain.Transaction, bc *blockchain.BlockChain) bool {
-	if ok, _ := bc.ValidTransaction(t); ok {
+	if ok, err := bc.ValidTransaction(t); ok {
 		p.set(t)
```

Comment:
Since the transactions are mapped by input hash, this type of situation currently would overwrite the existing transaction in the pool. Thats better than letting a double spend, but its not great. I'll work on it :+1: